scripts/forc-check.sh

```diff
@@ -5,14 +5,27 @@ main_dir=$(pwd)
 forc_fmt=$(realpath ./packages/forc/forc-binaries/forc-fmt)
 
 ERRORED=0
-
+RED='\033[0;31m'
+NC='\033[0m' # No Color
 for i in $forc_projects; do
-    cd ${i/Forc.toml/''}
-    eval $forc_fmt --check
+    cd "${i/Forc.toml/''}" || exit
+    eval "$forc_fmt" --check
     if [ $? = "1" ]; then
         ERRORED=1
     fi
-    cd $main_dir
+
+    # do authors checks only on projects, not on workspaces
+    if [ "$(head -n 1 Forc.toml)" != '[project]' ]; then
+        cd "$main_dir" || exit
+        continue
+    fi
+
+    authors=$(cat Forc.toml | grep authors)
+    if [ "$authors" != "authors = [\"Fuel Labs <contact@fuel.sh>\"]" ]; then
+        echo -e authors field should be: $RED"authors = [\"Fuel Labs <contact@fuel.sh>\"]"$NC but is $RED"$authors"$NC
+        ERRORED=1
+    fi
+    cd "$main_dir" || exit
```

Comment:
Maybe something like this could also work:

```sh
    # not tested / pseudo-code to illustrate

    author="authors = [\"Fuel Labs <contact@fuel.sh>\")]"
    is_invalid=$(grep -LE "/(\[project\])|(${author})/g" $i)

    if [ ! -z "${is_invalid}" ]; then
      echo "Authors field should be: $RED${author}"
      echo $i
      exit 1
    fi
```


Taken from [here](https://github.com/FuelLabs/fuels-ts/blob/cbd1459d08437de75da129d1cd46c30a1f33d675/scripts/tests-validate.sh#L1-L12).

Reply:
I stayed with the original solution because of the `authors` checking happening only on `[project]` files, which is IMO easier to understand off the bat with a split `if` statement for it. I did clean it up a bit, though.